MarsRover.Tests/MapParserShould.cs

```diff
@@ -9,26 +11,11 @@ public class MapParserShould
         [Fact]
         public void ReturnMap_GivenValidString()
         {
-            var squares = new List<Square>
-            {
-                new Square(SquareContent.Obstacle, 0, 0),
-                new Square(SquareContent.None, 1, 0),
-                new Square(SquareContent.None, 2, 0),
-                new Square(SquareContent.None, 0, 1),
-                new Square(SquareContent.None, 1, 1),
-                new Square(SquareContent.None, 2, 1),
-                new Square(SquareContent.None, 0, 2),
-                new Square(SquareContent.None, 1, 2),
-                new Square(SquareContent.None, 2, 2),
-                new Square(SquareContent.None, 0, 3),
-                new Square(SquareContent.None, 1, 3),
-                new Square(SquareContent.None, 2, 3)
-            };
-            var expected = new Map(3, 4, squares);
+            var result = MapParser.ParseMap("ONNN\nNNNN\nNNNN");
 
-            var result = MapParser.ParseMap("3,4\nONNN\nNNNN\nNNNN");
-
-            Assert.Equal(expected, result);
+            Assert.Equal(12, result.Squares.Count());
+            Assert.IsType<Map>(result);
+            // TODO: how to count squares with Obstacles without violating the Law of Demeter??

```

Comment:
Can you remember one strategy that I've mentioned before that you can use?
Alternatively, have you considered using a helper method here to check if the actual map is equal to the expected map?